### PR TITLE
Revise SoftPlus operation reference implementation 37559

### PR DESCRIPTION
**Pull request by yding10**
### Details:
 - improve the input type error check
 - add invalid input error test case
 - add backend test case
 - add attribute test case
### Tickets:
 - 37559

### CODE REVIEW

File: docs/ops/activation/SoftPlus_4.md [hunk 1]
```diff
@@ -19,16 +19,19 @@ SoftPlus(x) = \left\{\begin{array}{r}
 \end{array}\right.
 \f]
 
-**Note**: For numerical stability the operation reverts to the linear function when `x > threshold` where `threshold` depends on *T* and
-is chosen in such a way that the difference between the linear function and exact calculation is no more than `1e-6`.
-The `threshold` can be calculated with the following formula where `alpha` is the number of digits after the decimal point,
-`beta` is maximum value of *T* data type:
+**Note**:
```
Comment (ilyachur) [inline docs/ops/activation/SoftPlus_4.md:22]: @lazarevevgeny Could you please take a look?

File: ngraph/core/reference/include/ngraph/runtime/reference/softplus.hpp [hunk 2]
```diff
@@ -16,7 +16,7 @@ namespace ngraph
             template <typename T>
             void softplus(const T* arg, T* out, size_t count)
             {
-                const T threshold = static_cast<T>(std::log(std::numeric_limits<T>::max()));
+                const T threshold = 20.0;
```
Comment (ilyachur) [inline ngraph/core/reference/include/ngraph/runtime/reference/softplus.hpp:19]: Why do you use hardcoded value (Why 20?)?

Comment (yding10) [reply ngraph/core/reference/include/ngraph/runtime/reference/softplus.hpp:19]: @ilyachur current implementation is not aligned with spec. threshold is a range that we can choose from, 20 is chosen for FP32 based on experiment, to align to the spec and cpu-plugin implementation.  for FP16, it is also working that no overflow will happen as calculation is based on FP32 and converted to FP16 at final stage.  So either we change the spec or change the code here to get it aligned.

File: docs/ops/activation/SoftPlus_4.md [hunk 3]
```diff
@@ -19,16 +19,19 @@ SoftPlus(x) = \left\{\begin{array}{r}
...
+The `threshold` is internally defined parameter, which can be calculated with the following formula where `alpha` is the number of digits after the decimal point, `beta` is maximum value of *T* data type:
 
 \f[
 -log(e^{10^{-\alpha}} - 1.0) < threshold < log(\beta)
 \f]
 
-For example, if *T* is `fp32`, `threshold` should be `20` or if *T* is `fp16`, `threshold` should be `11`.
+For example:
+If *T* is `fp32`, `threshold` range is `13.8 ~ 88.7`, in practice `threshold` is set to be `20`.
+If *T* is `fp16`, in practice `threshold` is also set to be `20`，which is also ok as the data is operated in `fp32` format and then converted to `fp16` at the end.
```
Comment (lazarevevgeny) [inline docs/ops/activation/SoftPlus_4.md:34]: This is not true. There are plugins which calculate the model in pure FP16, so for them the value `11` is correct.

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: docs/ops/activation/SoftPlus_4.md (hunk 1 above)
Comment (lazarevevgeny) [reply docs/ops/activation/SoftPlus_4.md:22]: I don't agree with these changes. @a-sidorova did analysis and provided a formulae to calculate the threshold based on the data type. We should use this approach and do not hardcode any constants in the specification.

Comment (yding10) [reply docs/ops/activation/SoftPlus_4.md:22]: ok, revert back. update after getting alignemnt by email.

File: docs/ops/activation/SoftPlus_4.md (hunk 3 above)
Comment (yding10) [reply docs/ops/activation/SoftPlus_4.md:34]: ok, revert back. update after getting alignemnt by email.

File: ngraph/core/reference/include/ngraph/runtime/reference/softplus.hpp (hunk 2 above)
Comment (yding10) [reply ngraph/core/reference/include/ngraph/runtime/reference/softplus.hpp]: revert back ..

File: ngraph/core/src/op/softplus.cpp [hunk 4]
```diff
@@ -29,6 +29,13 @@ bool op::v4::SoftPlus::visit_attributes(AttributeVisitor& visitor)
 void op::v4::SoftPlus::validate_and_infer_types()
 {
     NGRAPH_OP_SCOPE(v4_SoftPlus_validate_and_infer_types);
+    const element::Type& input_et = get_input_element_type(0);
```
Comment (ilyachur) [inline ngraph/core/src/op/softplus.cpp:32]: Can you add more tests (for visit_attributes)?
https://github.com/openvinotoolkit/openvino/tree/master/ngraph/test/visitors/op

Comment (yding10) [reply ngraph/core/src/op/softplus.cpp:32]: there is no attribute for this op.  the "threshold" we discussed before is an internal defined parameter, which is not exposed to user.  is it ok not to add test for visit_attributes?

**Review by lazarevevgeny — APPROVED** [review]

**Review by a-sidorova — APPROVED** [review]

File: ngraph/core/src/op/softplus.cpp (hunk 4 above)
Comment (ilyachur) [reply ngraph/core/src/op/softplus.cpp:32]: Yes we need to add tests for such operations too.

Example you can find here: https://github.com/openvinotoolkit/openvino/blob/master/ngraph/test/visitors/op/result.cpp

Comment (yding10) [reply ngraph/core/src/op/softplus.cpp:32]: thanks for reminder, I have added it, please help check it.

**Review by ilyachur — APPROVED** [review]

File: ngraph/core/src/op/softplus.cpp (hunk 4 above)
Comment (yding10) [reply ngraph/core/src/op/softplus.cpp:32]: @ilyachur is this PR ok to be merged into master?

File: ngraph/test/visitors/op/softplus.cpp [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+#include "ngraph/ngraph.hpp"
+#include "ngraph/opsets/opset1.hpp"
+#include "util/visitor.hpp"
+
+using namespace std;
+using namespace ngraph;
+using ngraph::test::NodeBuilder;
+using ngraph::test::ValueMap;
+
+TEST(attributes, softplus)
```
Comment (ilyachur) [inline ngraph/test/visitors/op/softplus.cpp:15]: Hi could you please rebase your PR to the latest master. And please use parametrized visitor API test which was introduced in this PR: https://github.com/openvinotoolkit/openvino/pull/6181

Comment (yding10) [reply ngraph/test/visitors/op/softplus.cpp]: done.

**Pull request merged**